tmt/base.py

```diff
@@ -2089,6 +2089,12 @@ def tests(
         # Used in: tmt run test --name NAME, tmt test ls NAME...
         cmd_line_names: List[str] = list(Test._opt('names', []))
 
+        # Validate each name in cmd_line_names
+        for name in cmd_line_names:
+            if not name.isprintable():
+                raw_name = repr(name)
+                raise tmt.utils.GeneralError(f"Name {raw_name} is invalid as it isn't printable.")
```

Comment:
* if it's possible to fit `repr()` call into the string, please, do so. These single-use-temporary variables are like pollution :) `f"Name {repr(name)} is invalid ..."`.
* note that there are also two other places where we should cover this, `Tree.plans()` and `Tree.stories()`, both use names from `_opt('names')`, they just store the value in differently named variables.

Reply:
> if it's possible to fit repr() call into the string ...

Updated.

> there are also two other places where we should cover this, Tree.plans() and Tree.stories()

Added.

@happz, thanks for your quick review, please help to take a look again, thanks!

Reply:
Np.

Names are covered, now I'd recommend reducing the amount of copy & paste needed, to reduce the maintenance burden. How about putting these lines into something like `sanitize_cli_names(List[str]) -> List[str]`, returning the list or raising an error? We might need to add more checks in the future, who knows, and there would be one place where to do it.

Reply:
Hi @happz, a function `sanitize_cli_names` is added according to your suggestion. Please help to review again, thanks!

Reply:
Thank you!